From 7da622e834bf7e8e2a6eec51fae53084b3f190ac Mon Sep 17 00:00:00 2001
From: Steffen Winterfeldt <wfeldt@opensuse.org>
Date: Wed, 20 Oct 2021 12:25:03 +0200
Subject: fix test suite for 32-bit architectures

---
 test/prepare_test | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/prepare_test b/test/prepare_test
index 17203a3..389b7aa 100755
--- a/test/prepare_test
+++ b/test/prepare_test
@@ -414,7 +414,9 @@ sub nr
   $n = ((int rand 0x1000000) & 0xffff00) << 8;
   $n += int((rand 0x1000000) / 11);
 
-  return $n;
+  # Limit the result to a 32 bit number but avoid '&' to ensure it works
+  # with 32 bit perl.
+  return $n % 2 ** 32;
 }
 
 
-- 
2.33.1

